packages/nuxt/src/pages/utils.ts

```diff
@@ -87,6 +88,11 @@ export async function generateRoutesFromFiles (files: ScannedFile[], options: Ge
     // Array where routes should be added, useful when adding child routes
     let parent = routes
 
+    if(segments[segments.length - 1].endsWith('.server')) {
```

Comment:
One note: We might need to refactor this in future to support multiple ending segments. (But any change can be deferred to that point.)

```suggestion
    if (segments[segments.length - 1].endsWith('.server')) {
```

Reply:
sure !